Saurabh/transaction list

saurabhalp opened the pull request:
## Transaction List

## Description
-added `up_Arrow.png` and `down_arrow,png` 
-created transaction list
 -`index.tsx` 
 -`styles.tsx` 
 -`constants.tsx` added constant file which has all the constants used in transaction list

## How to use
` import {TRANSACTION_TYPES} from "./src/UI/Components/TransactionList/constants";`

`const transactions = [
  {
    type: TRANSACTION_TYPES.PAYMENT_TO,
    recipient: 'Shivam',
    amount: '500',
    date: '12 july, 2024',
    time: '10:30 AM',
    status: 'succeed',
  },
  {
    type: TRANSACTION_TYPES.RECEIVED_FROM,
    recipient: 'Saurabh',
    amount: '300',
    date: '15 sept, 2024',
    time: '2:15 PM',
    status: 'failed',
  }
];
const App = () => {
  return (
    <SafeAreaView>
      <TransactionList transactions={transactions} />
    </SafeAreaView>
  );
};
export default App; `

## Preview
![image](https://github.com/user-attachments/assets/d17ea203-756a-4c70-a457-2aa6ad9576ce)

Jha2508 commented:
Code seems fine but : 
1. The components should be on different files. Let us have item and list as two different components. [ Single Responsibility Principle]
2. Use Icon from some react native icon library, we cannot upload everything to the repo directly

saurabhalp commented:
## Update
 - separated `transactionItem` and `transactionList`
 - separate types for `transactionItem` and `transactionList`
 - removed images
 - used libraray for icon
 - updated front

Aashirvaad-P commented:
Can you resolve the conflicts quickly.

Aashirvaad-P commented on android/app/build.gradle:
```diff
@@ -119,7 +119,8 @@ dependencies {
-
-        iconFontNames: [ 'MaterialIcons.ttf'],
+        iconFontNames: [ 'MaterialIcons.ttf','Feather.ttf' ],
+        // Add other icon fonts if you're using them
```
Just change this comment to something that does look like AI generated
